### DISCUSSION TITLE
Is there any way to optimize json serialization in Bun ?

### DISCUSSION THREAD

**Discussion by masfahru** [oven-sh/bun#3633] [General]
I have tested Bun with other backend frameworks using TechEmpower benchmark tools, i use 3 VMs with 8 vcpus / 8 RAM in cloud.

The results can be seen here: https://www.techempower.com/benchmarks/#section=test&shareid=a4b12278-96a0-4cee-b7dc-4a3501e4134a&test=composite

![image](https://github.com/oven-sh/bun/assets/92277627/574116b8-4706-4fdb-8d13-c68969d70b02)

As it is shown on the result, Bun served plaintext for 695,777 rps, but it served json for 179,632 rps.
is there any way to optimize json serialization in Bun ?

here is the code i use to serve plaintext and json:
```javascript
const headerPlainText = {
  headers: {
    'Content-Type': 'text/plain',
    Server: 'bun',
  }
}

const headerJSON = {
  headers: {
    Server: 'bun',
  }
}

const servePlainText = () => new Response('Hello, World!', headerPlainText)

const serveJSON = () => Response.json({ message: 'Hello, World!' }, headerJSON);
```

**Comment by Jarred-Sumner** [comment]
This might help a little https://github.com/oven-sh/bun/pull/3656

In the meantime, try `new Response(JSON.stringify(obj), headerJSON)` instead

**Reply by masfahru** [reply]
thank you, can't wait for the next release
